src/storage/src/source/sql_server/replication.rs

```diff
@@ -0,0 +1,337 @@
+    AsyncOutputHandle, OperatorBuilder as AsyncOperatorBuilder, PressOnDropButton,
+};
+use mz_timely_util::containers::stack::AccountedStackBuilder;
+use timely::container::CapacityContainerBuilder;
+use timely::dataflow::channels::pushers::Tee;
+use timely::dataflow::operators::{CapabilitySet, Concat, Map};
+use timely::dataflow::{Scope, Stream as TimelyStream};
+use timely::progress::{Antichain, Timestamp};
+
+use crate::source::RawSourceCreationConfig;
+use crate::source::sql_server::{
+    DefiniteError, ReplicationError, SourceOutputInfo, TransientError,
+};
+use crate::source::types::{SignaledFuture, SourceMessage, StackedCollection};
+
+/// Used as a partition ID to determint the worker that is responsible for
+/// reading data from SQL Server.
+///
+/// TODO(sql_server1): It's possible we could have different workers
+/// replicate different tables, if we're using SQL Server's CDC features.
+static REPL_READER: &str = "reader";
+
+pub(crate) fn render<G: Scope<Timestamp = Lsn>>(
+    scope: G,
+    config: RawSourceCreationConfig,
+    outputs: BTreeMap<GlobalId, SourceOutputInfo>,
+    source: SqlServerSource,
+) -> (
+    StackedCollection<G, (u64, Result<SourceMessage, DataflowError>)>,
+    TimelyStream<G, Infallible>,
+    TimelyStream<G, ReplicationError>,
+    PressOnDropButton,
+) {
+    let op_name = format!("SqlServerReplicationReader({})", config.id);
+    let mut builder = AsyncOperatorBuilder::new(op_name, scope);
+
+    let (data_output, data_stream) = builder.new_output::<AccountedStackBuilder<_>>();
+    let (_upper_output, upper_stream) = builder.new_output::<CapacityContainerBuilder<_>>();
+
+    // Captures DefiniteErrors that affect the entire source, including all outputs
+    let (definite_error_handle, definite_errors) =
+        builder.new_output::<CapacityContainerBuilder<_>>();
+
+    let (button, transient_errors) = builder.build_fallible(move |caps| {
+        let busy_signal = Arc::clone(&config.busy_signal);
+        Box::pin(SignaledFuture::new(busy_signal, async move {
+            let [data_cap_set, upper_cap_set, definite_error_cap_set]: &mut [_; 3] =
+                caps.try_into().unwrap();
+
+            // TODO(sql_server1): Run ingestions across multiple workers.
+            if !config.responsible_for(REPL_READER) {
+                return Ok::<_, TransientError>(());
+            }
+
+            let connection_config = source
+                .connection
+                .resolve_config(
+                    &config.config.connection_context.secrets_reader,
+                    &config.config,
+                    InTask::Yes,
+                )
+                .await?;
+            let (mut client, connection) =
+                mz_sql_server_util::Client::connect(connection_config).await?;
+            // TODO(sql_server1): Move the connection into its own future.
+            mz_ore::task::spawn(|| "sql_server-connection", async move { connection.await });
```

Comment:
What does the TODO refer to? It seems to be its own future already. Also the async block can be reduced to just `connection`: 

```suggestion
            mz_ore::task::spawn(|| "sql_server-connection", connection);
```